File: Go/src/global.go
package main

import (
	"fmt"
	"sync"
)

type Global struct{}

func (g *Global) Action1() {
	fmt.Printf("\n\r...Action1...")
}

func (g *Global) Action2() {
	fmt.Printf("\n\r...Action2...")
}

var (
	instance *Global
	once     sync.Once
)

func Shared() *Global {
	once.Do(func() {
		instance = &Global{}
	})
	return instance
}

File: Rust/src/global.rs

use std::sync::OnceLock;

pub struct Global;

impl Global {
    pub fn action1(&self) {
        print!("\n\r...Action1...");
    }

    pub fn action2(&self) {
        print!("\n\r...Action2...");
    }
}

static GLOBAL: OnceLock<Global> = OnceLock::new();

pub fn global_shared() -> &'static Global {
    GLOBAL.get_or_init(|| Global)
}